docs/references/fluxyaml-config-files.md

```diff
@@ -1,229 +1,268 @@
 
-    After invoking the updaters, `fluxd` will then commit and push the resulting modifications to the git repository.
+The commands will be run with the `--git-path` being processed as a
+working directory -- which is not necessarily the same directory in
+which the `.flux.yaml` file was found. [See below](#execution-context)
+for more details on the execution context in which commands are run.
 
-3. `fluxd` will ignore any other yaml files under that path (e.g. resource manifests).
+### Using patch-updated configuration
 
-Generators and updaters are intentionally independent, in case a matching updater cannot be provided. It is hard to
-create updaters for some factorization technologies (particularly Configuration-As-Code). To improve the situation,
-a separate configuration file variant (`patchedUpdated`) is provided, which will be described later on.
+A patch-updated configuration generates manifests using commands, and
+records updates in a patch file.
 
-### Execution context of commands
+For example, when an automated image upgrade is run, fluxd will do
+this:
 
-`generators` and `updaters` are run in a POSIX shell inside the Flux container. This means that the `command`s supplied
-should be available in the [Flux container image](https://github.com/fluxcd/flux/blob/master/docker/Dockerfile.flux).
-Flux currently includes `Kustomize` and basic Unix shell tools. If the tools in the Flux image are not sufficient for
-your use case, you can include new tools in your own Flux-based image or, if the tools are popular enough, Flux
-maintainers can add them to the Flux image (please create an issue). In the future (once [Ephemeral
-containers](https://github.com/kubernetes/kubernetes/pull/59416) are available), you will be able to specify an container
-image for each command.
+ 1. run the generator commands and parse the manifests;
+ 2. find the manifest that needs to be updated, and calculate the
+    patch to it that performs the update;
+ 3. record that patch in the patch file.
 
-The working directory (also known as CWD) of the `command`s executed from a `.flux.yaml` file will be set to the
-target path (`--git-path` entry) used when finding that `.flux.yaml` file.
+When syncing, fluxd will generate the manifests as usual, then apply
+all the patches that have been recorded in the patch file.
 
-For example, when using flux with `--git-path=staging` on a git repository with this structure:
+This is how a patch-updated `.flux.yaml` looks:
 
-```sh
-├── .flux.yaml
-├── staging/
-├──── [...]
-├── production/
-└──── [...]
+```yaml
+version: 1
+patchUpdated:
+  generators:
+  - command: jk generate ./index.js
+  patchFile: flux-patch.yaml
 ```
 
-The commands in `.flux.yaml` will be executed with their working directory set to `staging.`
+The `generators` field is explained just above. The `patchFile` field
+gives a path, relative to the `--git-path` value, in which to record
+patches. `fluxd` will create the file if necessary, and commit any
```

Comment:
Nitpick:
```suggestion
patches. `fluxd` will create the file when needed, and commit any
```

Reply:
Also, it may be worth mentioning that it's the user's responsibility to keep the patch file consistent with the generated manifests (e.g. if a manifests is removed, their patch counterpart needs to be removed too).

Reply:
Many users were confused about GC and patches. We should mention that even if Flux GC removes a deployment the patch will error out if not removed manually

Reply:
I've mentioned the need to manually remove patches when manifests go.